app/lib/actions/katello/sync_plan/destroy.rb

```diff
@@ -14,7 +14,9 @@ def plan(sync_plan)
 
         def finalize
           sync_plan = ::Katello::SyncPlan.find(input[:sync_plan][:id])
+          recurring_logic = sync_plan.recurring_logic
           sync_plan.destroy!
+          recurring_logic.destroy!
         end
```

Comment:
I think this logic could be moved to a :dependent => :destroy  on the association, and then this whole action isn't needed.